lib/internal/fs/promises.js

```diff
@@ -557,7 +558,7 @@ async function readFileHandle(filehandle, options) {
 
     if (bytesRead === 0 ||
         totalRead === size ||
-        (bytesRead !== buffer.length && !chunkedRead)) {
+        (bytesRead !== buffer.length && !chunkedRead && !noSize)) {
```

Comment:
What about changing the chunkedRead above instead:

```js
const chunkedRead = length > kReadFileBufferLength || size === 0;
```

That would automatically check for it and it's one boolean check in the loop less.

Reply:
I believe the premise here for `chunkedRead ` is already knowing the `size` of the file, meaning `size === 0` cannot imply the need for `chunkedRead`, have I misunderstand something?